osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -156,6 +156,10 @@ public virtual string SettingDescription
         [JsonIgnore]
         public virtual bool ValidForMultiplayerAsFreeMod => true;
 
+        /// <inheritdoc/>
+        [JsonIgnore]
+        public virtual bool AlwaysValidForSubmission => false;
```

Comment:
This is added for osu-web's sake. See: https://github.com/ppy/osu-web/pull/10704.

As per https://github.com/ppy/osu-web/pull/10704#discussion_r1380875900, wasn't sure whether to go with flag or hardcode in the end, so I picked flag. Will change if consensus from both sides swings the other way.